flink-clients/src/main/java/org/apache/flink/client/cli/CliFrontend.java

```diff
@@ -704,7 +708,7 @@ protected void executeProgram(final Configuration configuration, final PackagedP
 	 *
 	 * @return A PackagedProgram (upon success)
 	 */
-	PackagedProgram buildProgram(final ProgramOptions runOptions)
+	PackagedProgram buildProgram(final ProgramOptions runOptions, boolean setConfiguration)
```

Comment:
Is this additional parameter only introduced for tests? In production code I can only see `false` as a parameter. If yes, then we should find a different solution because we should not introduce code in production code that is only used in tests.

Reply:
fixed